src/track/globaltrackcache.cpp

```diff
@@ -708,8 +706,12 @@ void GlobalTrackCache::evictAndSave(
     DEBUG_ASSERT(!isCached(cacheEntryPtr->getPlainPtr()));
     saveEvictedTrack(cacheEntryPtr->getPlainPtr());
 
-    // here the cacheEntryPtr goes out of scope, the cache entry is
-    // deleted including the owned track
+    // Explicitly release the cacheEntryPtr including the owned
+    // track object while the cache is still locked.
+    cacheEntryPtr = {};
```

Comment:
This is new to me. I like to under stand it, and like to know what the advantages are: 

Is it:  
std::shared_ptr = std::initalizer_list<T>() 

What is T in this case? 

std::shared_ptr has no std::initalizer_list constructor, so i think the default constructor is picked. 

This means the line can reads: 
Correct? 

cacheEntryPtr = GlobalTrackCacheEntryPointer();

I think I would prefer: 

cacheEntryPtr.reset() 

This is easy to read and we bypass a lot of compiler magic.

Reply:
Reassignment with default constructed object without literally repeating the type name. Just because I forgot about the reset() method that is applicable here. Fixed.

Reply:
Thank you.